utils/opencas.py

```diff
@@ -689,3 +736,71 @@ def stop(flush):
 
     error.raise_nonempty()
 
+
+def get_devices_state():
+    device_list = get_caches_list()
+
+    devices = {"core_pool": [], "caches": {}, "cores": {}}
+
+    core_pool = False
+    prev_cache_id = -1
+
+    for device in device_list:
+        if device["type"] == "core pool":
+            core_pool = True
+            continue
+        if device["type"] == "cache":
+            core_pool = False
+            prev_cache_id = int(device["id"])
+            devices["caches"].update(
+                {
+                    int(device["id"]): {
+                        "device": device["disk"],
+                        "status": device["status"],
+                    }
+                }
+            )
+        elif device["type"] == "core":
+            core = {"device": device["disk"], "status": device["status"]}
+            if core_pool:
+                devices["core_pool"].append(core)
+            else:
+                core.update({"cache_id": prev_cache_id})
+                devices["cores"].update({int(device["id"]): core})
+
+    return devices
+
+
+def wait_for_startup(timeout=300, interval=4, wait_for_lazy=False):
+    try:
+        config = cas_config.from_file(
+            cas_config.default_location, allow_incomplete=True
+        )
+    except Exception as e:
+        raise Exception("Unable to load opencas config. Reason: {0}".format(str(e)))
+
+    stop_time = time.time() + int(timeout)
+
+    not_initialized = None
+    target_core_state = config.get_startup_cores()
```

Comment:
This variable and ` config.get_startup_cores()` are unneeded.

Reply:
I think it is used, we need it to check if we've arrived at desired configuration